src/Workspaces/Remote/ServiceHub/Shared/ClientDirectStream.cs

```diff
@@ -9,100 +9,114 @@
-        {
-            get { return _stream.ReadTimeout; }
-            set { _stream.ReadTimeout = value; }
-        }
-
-        public override int WriteTimeout
-        {
-            get { return _stream.WriteTimeout; }
-            set { _stream.WriteTimeout = value; }
-        }
-
-        public override bool CanRead => _stream.CanRead;
-        public override bool CanSeek => _stream.CanSeek;
-        public override bool CanWrite => _stream.CanWrite;
-        public override long Length => _stream.Length;
-        public override bool CanTimeout => _stream.CanTimeout;
-
-        public override void Flush() => _stream.Flush();
-        public override Task FlushAsync(CancellationToken cancellationToken) => _stream.FlushAsync(cancellationToken);
-
-        public override long Seek(long offset, SeekOrigin origin) => _stream.Seek(offset, origin);
-        public override void SetLength(long value) => _stream.SetLength(value);
-
-        public override int ReadByte() => _stream.ReadByte();
-        public override void WriteByte(byte value) => _stream.WriteByte(value);
-
-        public override int Read(byte[] buffer, int offset, int count) => _stream.Read(buffer, offset, count);
-        public override void Write(byte[] buffer, int offset, int count) => _stream.Write(buffer, offset, count);
-
-        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) => _stream.ReadAsync(buffer, offset, count, cancellationToken);
-        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) => _stream.WriteAsync(buffer, offset, count, cancellationToken);
-
-        public override IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallback callback, object state) => _stream.BeginRead(buffer, offset, count, callback, state);
-        public override int EndRead(IAsyncResult asyncResult) => _stream.EndRead(asyncResult);
-
-        public override IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback callback, object state) => _stream.BeginWrite(buffer, offset, count, callback, state);
-        public override void EndWrite(IAsyncResult asyncResult) => _stream.EndWrite(asyncResult);
-
-        public override void Close() => _stream.Close();
-
-        public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken) => _stream.CopyToAsync(destination, bufferSize, cancellationToken);
-
-        protected override void Dispose(bool disposing) => _stream.Dispose();
-
-        public override object InitializeLifetimeService()
-        {
-            throw new NotSupportedException();
-        }
-
-        public override ObjRef CreateObjRef(Type requestedType)
+        private class ClientDirectStream : Stream
         {
-            throw new NotSupportedException();
+            // 4KB buffer size
+            private const int BUFFERSIZE = 4 * 1024;
```

Comment:
based on some testing. server side as I increase buffer size, time reduced, client side didn't matter that much. my guess is reading taking 4K at a time to buffer is good enough. 

probably keep get tweaked as I do more detail perf measure